include/request.h

```diff
@@ -3,22 +3,86 @@
 
 #include <string>
 #include <vector>
-#include "header.h"
-
-struct Request {
-  std::string method;
-  // handler_path + file_path is path from request, to decide which handler to use
-  // Example: /pizza/crust/cheese.jpg, 
-  //          handler_path = "/pizza"
-  //          file_path = "/crust/cheese.jpg"
-  std::string handler_path; 
-  std::string file_path;
-  std::string base_path;
-  std::string uri;
-  int http_version_major;
-  int http_version_minor;
-  std::vector<Header> headers;
-  std::string raw_request; // the original request text
+#include <memory>
+
+// Represents an HTTP Request.
+//
+// Usage:
+//   auto request = Request::Parse(raw_request);
+class Request {
+  public:
+    static std::unique_ptr<Request> Parse(const std::string& raw_request);
+
+    std::string raw_request() const;
+    std::string method() const;
+    std::string uri() const;
+    std::string version() const;
+
+    using Headers = std::vector<std::pair<std::string, std::string>>;
+    Headers headers() const;
+
+    std::string body() const;
+
+    // For testing. Must be removed to comply with API
+    enum result_type { good, bad, indeterminate };
+
+  private:
+
+
+    // Handle the next character of input.
+    result_type consume(char input);
+
+    std::string m_raw_request;
```

Comment:
change to raw_request_ (trailing underscore). same for the rest of the member variables